gsoc_final_post/inigo_tellaetxe_gsoc_2024_final_report.rst

```diff
@@ -0,0 +1,169 @@
+--------------------
+
+* **Literature Review on synthetic tractography generation using AutoEncoders**. Main inspirations: 
+
+  * The `FINTA <https://10.1016/j.media.2021.102126>`_ and `GESTA <https://doi.org/10.1016/j.media.2023.102761>`_ papers because they provide a relatively simple AE architecture with an open source sampling algorithm, easy to reuse.
+
+  * `Variational AutoEncoders for Regression <https://doi.org/10.1007/978-3-030-32245-8_91>`_, which provided a good starting point for conditional Variational AutoEncoders with direct application to brain aging, related to the project's objectives.
+
+  * `Attribute-based Regularization of latent spaces for variational AutoEncoders <https://doi.org/10.1007/s00521-020-05270-2>`_ and `Adversarial AutoEncoders <http://arxiv.org/abs/1511.05644>`_, which inspired the use of Attribute-based regularization for conditioning on a continuous variable (age); and the use of adversarial training for conditioning on categoric variables (fiber bundle or neurodegeneration status), respectively.
+
+* **Replicated the** `GESTA <https://doi.org/10.1016/j.media.2023.102761>`_ **architecture originally implemented in PyTorch (vanilla AutoEncoder, not variational) using TensorFlow2+Keras.** 
+  Validated the results using the FiberCup dataset. The model is found in the ``ae_model.py`` module of the `TractoEncoder GSoC <https://github.com/itellaetxe/tractoencoder_gsoc>`_ repository. The architecture can be summarized as in the following image:
+
+  .. image:: https://github.com/dipy/dipy.org/blob/master/_static/images/gsoc/2024/inigo/inigo_vanilla_autoencoder.png
+    :alt: AE architecture diagram.
+    :align: center
+    :width: 800
+
+  * Weight and bias initializers are different in PyTorch compared to TensorFlow2, so the PyTorch behavior was replicated using custom initializers. Different weight & bias initialization strategies can lead to drastically different training results, so this step took extra care.
+  * The upsampling layers of the Decoder block use linear interpolation in PyTorch by default, whereas in TensorFlow2, there is no native implementation for this, and nearest neighbor (NN) interpolation is used instead. This is a significant difference in implementations, and to replicate the PyTorch behavior a custom linear interpolating upsampling layer was implemented in TF2. However, after training with both NN and linear interpolation, the results were very similar, so the custom layer was not used in the final implementation.
+  * The figure below shows a summary of the replication results:
+
+    .. image:: https://github.com/dipy/dipy.org/blob/master/_static/images/gsoc/2024/inigo/fibercup_replicated.png
+      :alt: Replication of the GESTA architecture results on the FiberCup dataset.
+      :align: center
+      :width: 800
+
+
+* **Implemented a Variational AutoEncoder (VAE) architecture based on the** `GESTA <https://doi.org/10.1016/j.media.2023.102761>`_ **architecture.**
```

Comment:
Same as above: are we sure this adds the additional loss terms? If not, this would again be FINTA. I do not remember having discussed about this.

Reply:
True, I wanted to say FINTA, again.

I would say that the majority of the architecture is the same as FINTA. I am only adding 2 things to give FINTA the variational regularization:
- Two dense layers that output $\mu_z$ and $log{\sigma^2_z}$
- The KL divergence loss component

Thus, I would say that it is based on FINTA, even if it is slightly modified

Reply:
:+1:.